### PR TITLE
Consolidate addCommentAdvanced into addComment

### PR DESCRIPTION
**Pull request by ayashjorden**
Fixes jira-node/node-jira-client/issues/259

### CODE REVIEW

**Comment by ayashjorden** [comment]
Hi @pioug ,
Appreciate your feedback on this one as its a priority for my team.
The latest release does not include the `addCommentAdvanced` or the ability to pass `addComment` with comment options object.

Thank you

File: package.json [hunk 1]
```diff
@@ -1,13 +1,14 @@
 {
   "name": "jira-client",
-  "version": "6.21.1",
+  "version": "6.22.0",
```
Comment (pioug) [inline package.json:3]: Bumping the package version is out-of-the-scope. I mean, you don't know what the next version should be, neither do I.

File: package.json [hunk 2]
```diff
@@ -33,7 +34,7 @@
   "scripts": {
     "build": "rm -rf lib && babel src --out-dir lib",
     "docs-build": "rm -rf docs && esdoc -c esdoc.json",
-    "lint": "eslint ./",
+    "lint": "eslint --fix ./",
```
Comment (pioug) [inline package.json:37]: Do not change the scripts, it's out of scope

**Comment by pioug** [comment]
I am not sure to understand the goal of your PR. Is it just a refactoring? Seems like `addCommentAdvanced` can already achieve what you need. I supposed `addComment` has been preserved to avoid any breaking change so removing it now would be just troublesome.

`addCommentAdvanced` is included and seems to be documented https://jira-node.github.io/class/src/jira.js~JiraApi.html#instance-method-addCommentAdvanced 🤔  

Can you explain again what you are aiming for?

**Comment by ayashjorden** [comment]
My change takes into account both input options (simple string comment and comment options), so its actually improves compatibility over having two methods that are doing the same.
`addCommentAdvanced` might be ambiguous to the consumer (even though its documented).

I've updated my commit based on your feedback, LMK what you think.

File: package.json (hunk 2 above)
Comment (ayashjorden) [reply package.json]: Fixed

File: package.json (hunk 1 above)
Comment (ayashjorden) [reply package.json]: MMM, then who?

**Comment by pioug** [comment]
I am not satisfied the API that you proposed either. An argument that can be either a string or object is even more ambiguous than having `addComment` and `addCommentAdvanced`. So considering your change as an "improvement" is arguable.

Can you explain why is this a priority for your team? Maybe I miss something. From my point of view, it seems like nitpicking on method names. I can't reasonably accept PR every time someone is not happy with them.

I am not the original maintainer so I don't want to break the API for any reason. If I have too, it must really worth it.

**Comment by ayashjorden** [comment]
Probably missed the "advanced" method in vscode intellisense before I looked in the codebase, thought this would make things clearer.

If you're not the maintainer and don't feel comfortable with it, we can drop it since I have an advanced solution now. (pun partially intended).
Thank you

**Comment by pioug** [comment]
Thank you very much understanding 👌 There is definitely something we could do to help people finding the method `addCommentAdvanced` more easily.

**Pull request closed without merging**